diff --git a/Makefile b/Makefile
index ca5d392..9fe3f67 100644
--- a/Makefile
+++ b/Makefile
@@ -215,6 +215,7 @@ CFLAGS = -DPD -DHAVE_UNISTD_H -DUSEAPI_DUMMY \
 LDFLAGS += $(ADDITIONAL_LDFLAGS)
 CSHARP_LDFLAGS += $(ADDITIONAL_LDFLAGS)
 JAVA_LDFLAGS += $(ADDITIONAL_LDFLAGS)
+include extra.make
 
 .PHONY: libpd csharplib javalib javadoc javasrc install uninstall clean clobber
 
@@ -226,10 +227,10 @@ else
 endif
 
 $(LIBPD): ${PD_FILES:.c=.o} ${UTIL_FILES:.c=.o} ${EXTRA_FILES:.c=.o}
-	$(CC) -o $(LIBPD) $^ $(LDFLAGS) -lm -lpthread
+	$(CC) -o $(LIBPD) $^ $(EXTRA_OBJS) $(LDFLAGS) -lm -lpthread $(EXTRA_LIBS)
 
 $(LIBPD_STATIC): ${PD_FILES:.c=.o} ${UTIL_FILES:.c=.o} ${EXTRA_FILES:.c=.o}
-	ar rcs $(LIBPD_STATIC) $^
+	ar rcs $(LIBPD_STATIC) $^ $(EXTRA_OBJS)
 
 javalib: $(JNIH_FILE) $(PDJAVA_JAR)
 
diff --git a/extra.make b/extra.make
new file mode 100644
index 0000000..6036c82
--- /dev/null
+++ b/extra.make
@@ -0,0 +1,13 @@
+PD_EXTRA_FILES += $(patsubst %, ../pdxtra/%.c, \
+muse chrd rand rind ntof fton flenc fldec slx sly same ceil delp linp linp~ \
+radix 0x21 0x21~ x is has pak unpak 0x40pak 0x40unpak chrono tabread2~ tabosc2~ \
+myqwil blunt/hotop blunt/revop blunt/blunt)
+
+PD_EXTRA_FILES += $(patsubst %, ../pd-cyclone/cyclone_objects/binaries/audio/%.c, \
+curve lessthan greaterthan triangle)
+
+CFLAGS += -I../pd-cyclone/shared
+
+EXTRA_OBJS :=
+
+EXTRA_LIBS :=
diff --git a/libpd_wrapper/z_libpd.c b/libpd_wrapper/z_libpd.c
index 6bf7479..8fd9e64 100644
--- a/libpd_wrapper/z_libpd.c
+++ b/libpd_wrapper/z_libpd.c
@@ -52,6 +52,11 @@ int sys_pollgui(void);
   void pique_setup(void);
   void sigmund_tilde_setup(void);
   void stdout_setup(void);
+  void myQwil_setup(void);
+  void curve_tilde_setup(void);
+  void lessthan_tilde_setup(void);
+  void greaterthan_tilde_setup(void);
+  void triangle_tilde_setup(void);
 #endif
 
 static PERTHREAD t_atom *s_argv = NULL;
@@ -103,6 +108,11 @@ int libpd_init(void) {
   pique_setup();
   sigmund_tilde_setup();
   stdout_setup();
+  myQwil_setup();
+  curve_tilde_setup();
+  lessthan_tilde_setup();
+  greaterthan_tilde_setup();
+  triangle_tilde_setup();
 #endif
 #ifndef LIBPD_NO_NUMERIC
   setlocale(LC_NUMERIC, "C");
