diff --git a/Makefile b/Makefile
index 180c9ff..d60a0f3 100644
--- a/Makefile
+++ b/Makefile
@@ -261,9 +261,8 @@ install:
 	install -m 644 libpd_wrapper/z_libpd.h $(includedir)/libpd
 	install -m 644 pure-data/src/m_pd.h $(includedir)/libpd
 	if [ -e libpd_wrapper/util/z_queued.o ]; then \
-	  install -d $(includedir)/libpd/util; \
-	  install -m 644 libpd_wrapper/util/z_print_util.h $(includedir)/libpd/util; \
-	  install -m 644 libpd_wrapper/util/z_queued.h $(includedir)/libpd/util; \
+	  install -m 644 libpd_wrapper/util/z_print_util.h $(includedir)/libpd; \
+	  install -m 644 libpd_wrapper/util/z_queued.h $(includedir)/libpd; \
 	  install -m 644 cpp/*hpp $(includedir)/libpd; \
 	fi
 	install -d $(libdir)
