diff --git a/Makefile b/Makefile
index ca5d392..90a2188 100644
--- a/Makefile
+++ b/Makefile
@@ -215,6 +215,7 @@ CFLAGS = -DPD -DHAVE_UNISTD_H -DUSEAPI_DUMMY \
 LDFLAGS += $(ADDITIONAL_LDFLAGS)
 CSHARP_LDFLAGS += $(ADDITIONAL_LDFLAGS)
 JAVA_LDFLAGS += $(ADDITIONAL_LDFLAGS)
+include extra.make
 
 .PHONY: libpd csharplib javalib javadoc javasrc install uninstall clean clobber
 
diff --git a/extra.make b/extra.make
new file mode 100644
index 0000000..53855dd
--- /dev/null
+++ b/extra.make
@@ -0,0 +1,8 @@
+PD_EXTRA_FILES += $(patsubst %, ../pd-quilt/src/%.c, \
+0x40paq 0x40unpaq blunt chrd chrono delp fldec flenc fton has is linp linp~ \
+metro~ muse ntof paq radix rand rind same slx sly tabosc2~ tabread2~ unpaq all/quilt)
+
+PD_EXTRA_FILES += $(patsubst %, ../pd-cyclone/cyclone_objects/binaries/audio/%.c, \
+curve lessthan greaterthan)
+
+CFLAGS += -I../pd-cyclone/shared
diff --git a/libpd_wrapper/z_libpd.c b/libpd_wrapper/z_libpd.c
index 6bf7479..1fa73c5 100644
--- a/libpd_wrapper/z_libpd.c
+++ b/libpd_wrapper/z_libpd.c
@@ -52,6 +52,10 @@ int sys_pollgui(void);
   void pique_setup(void);
   void sigmund_tilde_setup(void);
   void stdout_setup(void);
+  void quilt_setup(void);
+  void curve_tilde_setup(void);
+  void lessthan_tilde_setup(void);
+  void greaterthan_tilde_setup(void);
 #endif
 
 static PERTHREAD t_atom *s_argv = NULL;
@@ -103,6 +107,10 @@ int libpd_init(void) {
   pique_setup();
   sigmund_tilde_setup();
   stdout_setup();
+  quilt_setup();
+  curve_tilde_setup();
+  lessthan_tilde_setup();
+  greaterthan_tilde_setup();
 #endif
 #ifndef LIBPD_NO_NUMERIC
   setlocale(LC_NUMERIC, "C");
